# origin: Debian (herbert)
# cset: n/a
# inclusion: probably not suitable for mainline (yet)
# description: modular vesafb
# revision date: 2004-09-04

diff -urN kernel-source-2.4.26/drivers/video/vesafb.c kernel-source-2.4.26-1/drivers/video/vesafb.c
--- kernel-source-2.4.26/drivers/video/vesafb.c	2003-11-29 05:26:21.000000000 +1100
+++ kernel-source-2.4.26-1/drivers/video/vesafb.c	2003-11-29 20:53:45.000000000 +1100
@@ -696,6 +696,13 @@
 	return 0;
 }
 
+static void __exit vesafb_exit(void)
+{
+	unregister_framebuffer(&fb_info);
+	iounmap(video_vbase);
+	release_mem_region(video_base, video_size);
+}
+
 /*
  * Overrides for Emacs so that we follow Linus's tabbing style.
  * ---------------------------------------------------------------------------
@@ -705,3 +712,7 @@
  */
 
 MODULE_LICENSE("GPL");
+#ifdef MODULE
+module_init(vesafb_init);
+#endif
+module_exit(vesafb_exit);
diff -urN kernel-source-2.4.26/drivers/video/Config.in kernel-source-2.4.26-1/drivers/video/Config.in
--- kernel-source-2.4.26/drivers/video/Config.in	2004-02-19 00:36:31.000000000 +1100
+++ kernel-source-2.4.26-1/drivers/video/Config.in	2004-02-22 20:28:28.000000000 +1100
@@ -91,7 +91,7 @@
       tristate '  TGA framebuffer support' CONFIG_FB_TGA
    fi
    if [ "$CONFIG_X86" = "y" ]; then
-      bool '  VESA VGA graphics console' CONFIG_FB_VESA
+      tristate '  VESA VGA graphics console' CONFIG_FB_VESA
       tristate '  VGA 16-color graphics console' CONFIG_FB_VGA16
       tristate '  Hercules mono graphics console (EXPERIMENTAL)' CONFIG_FB_HGA
       define_bool CONFIG_VIDEO_SELECT y
